forum/qa-plugin/report-reason/qa-plugin.php

```diff
@@ -0,0 +1,117 @@
+    Plugin Minimum Question2Answer Version: 1.7
+    Plugin Update Check URI: 
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    More about this license: http://www.gnu.org/licenses/gpl.html
+    
+*/
+
+if(!defined('QA_VERSION'))
+{
+    header('Location: ../../');
+    exit;
+}
+
+// language file
+qa_register_plugin_phrases('q2apro-flag-reasons-lang-*.php', 'q2apro_flagreasons_lang');
+
+// page
+qa_register_plugin_module('page', 'q2apro-flag-reasons-page.php', 'q2apro_flag_reasons_page', 'q2apro flag reasons Page');
+
+// layer 
+qa_register_plugin_layer('q2apro-flag-reasons-layer.php', 'q2apro flag reasons layer');
+
+// admin
+qa_register_plugin_module('module', 'q2apro-flag-reasons-admin.php', 'q2apro_flagreasons_admin', 'q2apro flag reasons Admin');
+
+// track events
+qa_register_plugin_module('event', 'q2apro-flag-reasons-event.php','q2apro_flagreasons_event','q2apro flag reasons Event');
+
+
+
+function q2apro_flag_reasonid_to_readable($reasonId)
+{
+    /*
+    Reasonids for Flags:
+    1 - spam
+    2 - quality
+    3 - rude
+    4 - edit
+    5 - duplicate
+    6 - migrate
+    */
+    
+    switch($reasonId)
+    {
+        case 1:
+            return qa_lang('q2apro_flagreasons_lang/reason_quality');
+            break;
+        case 2:
+            return qa_lang('q2apro_flagreasons_lang/reason_spam');
+            break;
+        case 3:
+            return qa_lang('q2apro_flagreasons_lang/reason_rude');
+            break;
+        case 4:
+            return qa_lang('q2apro_flagreasons_lang/reason_edit');
+            break;
+        case 5:
+            return qa_lang('q2apro_flagreasons_lang/reason_duplicate');
+            break;
+        case 6:
+            return qa_lang('q2apro_flagreasons_lang/reason_migrate');
+            break;
+        default: 
+            return '';
+    }
+}
+
+function q2apro_get_postflags($postId)
+{
+    return qa_db_read_all_assoc( qa_db_query_sub('
+            SELECT userid, postid, reasonid, notice 
+            FROM ^flagreasons
+            WHERE postid = #
+            ', $postId
+            ));
+}
+
+function q2apro_count_postflags_output($postId)
+{
+    $flags = q2apro_get_postflags($postId);
+    
+    $flagOutput = '';
+    
+    // count reasons
+    foreach($flags as $flag)
+    {
+        $handle = qa_userid_to_handle($flag['userid']);
+        $flagOutput .= (empty($flagoutput) ? '' : '<br>');
+        $flagOutput .= '✌ '.q2apro_flag_reasonid_to_readable($flag['reasonid']).' ('.$handle;
```

Comment:
czemu tu jest ten znaczek?

Reply:
To jakoś miało się wyświetlać w ramce z powodem zgłoszenia. Obecnie jest taki ciemny theme, że można to faktycznie wyrzucić